File: 2025_06/leetcode2025-06-10.cpp
#include <iostream>
#include <string>
#include <climits>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxDifference(string s) {
        unordered_map<char, int> cnt;
        for (auto c : s){
            cnt[c]++;
        }
        int mx = 0, mn = INT_MAX;
        for (auto& kv : cnt){
            if (kv.second & 1){
                mx = max(mx, kv.second);
            }
            else{
                mn = min(mn, kv.second);
            }
        }
        return mx-mn;
    }
};

int main(){
    Solution sl;
    string s = "aaaaabbc";
    cout << sl.maxDifference(s) << endl;
}

File: leetcode2025/2025_06/leetcode2025-06-10.py
from collections import Counter

class Solution:
    def maxDifference(self, s: str) -> int:
        cnt = Counter(s)
        oddMx, evenMn = 0, float("inf")
        for c, v in cnt.items():
            if v&1:
                oddMx = max(oddMx, v)
            else:
                evenMn = min(evenMn, v)
        return oddMx-evenMn

if __name__ == "__main__":
    sl = Solution()
    s = "aaaaabbc"
    print(sl.maxDifference(s))
